src/app/modules/submission/encrypt-submission/encrypt-submission.controller.ts

```diff
@@ -94,286 +85,258 @@ const Payment = getPaymentModel(mongoose)
-        error: formResult.error,
-      })
-      const { errorMessage, statusCode } = mapRouteError(formResult.error)
-      return res.status(statusCode).json({ message: errorMessage })
-    }
+  setFormTags(formDef)
 
-    setFormTags(formResult.value)
+  const form = req.body.formsg.encryptedFormDef
+  if (!form) return res.send(new EncryptedFormDefinitionNotRetrievedError())
 
-    const checkFormIsEncryptModeResult = checkFormIsEncryptMode(
-      formResult.value,
-    )
-    if (checkFormIsEncryptModeResult.isErr()) {
-      logger.error({
-        message:
-          'Trying to submit non-encrypt mode submission on encrypt-form submission endpoint',
-        meta: logMeta,
-      })
-      const { statusCode, errorMessage } = mapRouteError(
-        checkFormIsEncryptModeResult.error,
-      )
-      return res.status(statusCode).json({
-        message: errorMessage,
-      })
-    }
-    const form = checkFormIsEncryptModeResult.value
+  const ensurePipeline = new Pipeline(
+    ensurePublicForm,
+    ensureValidCaptcha,
+    ensureFormWithinSubmissionLimits,
+  )
 
-    const ensurePipeline = new Pipeline(
-      ensurePublicForm,
-      ensureValidCaptcha,
-      ensureFormWithinSubmissionLimits,
-    )
+  const hasEnsuredAll = await ensurePipeline.execute({
+    form,
+    logMeta,
+    req,
+    res,
+  })
 
-    const hasEnsuredAll = await ensurePipeline.execute({
-      form,
-      logMeta,
-      req,
-      res,
-    })
+  if (!hasEnsuredAll) return
 
-    if (!hasEnsuredAll) {
-      return
-    }
+  const encryptedPayload = req.body.formsg.encryptedPayload
+  if (!encryptedPayload) return res.send(new EncryptedPayloadNotFoundError())
 
-    // Create Incoming Submission
-    const { encryptedContent, responses, responseMetadata, paymentProducts } =
-      req.body
-    const incomingSubmissionResult = IncomingEncryptSubmission.init(
-      form,
-      responses,
-      encryptedContent,
+  // Create Incoming Submission
+  const { encryptedContent, responses, responseMetadata, paymentProducts } =
+    encryptedPayload
+  const incomingSubmissionResult = IncomingEncryptSubmission.init(
+    form,
+    responses,
+    encryptedContent,
+  )
+  if (incomingSubmissionResult.isErr()) {
+    const { statusCode, errorMessage } = mapRouteError(
+      incomingSubmissionResult.error,
     )
-    if (incomingSubmissionResult.isErr()) {
-      const { statusCode, errorMessage } = mapRouteError(
-        incomingSubmissionResult.error,
-      )
-      return res.status(statusCode).json({
-        message: errorMessage,
-      })
-    }
-    const incomingSubmission = incomingSubmissionResult.value
+    return res.status(statusCode).json({
+      message: errorMessage,
+    })
+  }
+  const incomingSubmission = incomingSubmissionResult.value
 
-    delete (req.body as SetOptional<EncryptSubmissionDto, 'responses'>)
-      .responses
+  // delete (req.body as SetOptional<EncryptSubmissionDto, 'responses'>)
+  //   .responses
```

Comment:
dangling comment

Reply:
Thanks! Removed in [b5080a5](https://github.com/opengovsg/FormSG/pull/6595/commits/b5080a5278ce7edfe139962b6b9a7ba9b9402e4f)